docs/dev/hacking-guide.md

```diff
@@ -0,0 +1,233 @@
+
+
+Second step, similar to running the component locally we need to scale down CVO.
+
+```
+oc scale --replicas=0 deploy/cluster-version-operator -n openshift-cluster-version
+```
+Next, edit machine-api-operator-images ConfigMap and place a link to your image there.
+
+The openshift-machine-api project contains all the resources and components that are used by the Machine API. There is a ConfigMap named machine-api-operator-images. This ConfigMap contains references to all the images used by the Machine API Operator, it uses these to deploy the controller and ensure they are the proper images. The ConfigMap looks something like this:
+```
+apiVersion: v1
+kind: ConfigMap
+data:
+  images.json: |
+    {
+        "machineAPIOperator": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "clusterAPIControllerAWS": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "clusterAPIControllerOpenStack": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "clusterAPIControllerLibvirt": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "clusterAPIControllerBareMetal": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "clusterAPIControllerAzure": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "clusterAPIControllerGCP": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "clusterAPIControllerOvirt": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "clusterAPIControllerVSphere": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "baremetalOperator": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "baremetalIronic": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "baremetalIronicInspector": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "baremetalIpaDownloader": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "baremetalMachineOsDownloader": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "baremetalStaticIpManager": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:..."
+    }
+```
+
+```
+oc edit configmap/machine-api-operator-images -n openshift-machine-api
+```
+
+With the new image information loaded into the ConfigMap, the next thing you might do is replace the Machine API operator. This operator controls how the specific cloud controllers are deployed and coordinated. You only change this component if there is something you are testing.
+
+The easiest way to change this operator is to scale down the deployment, change the image reference in the deployment, and then scale it back up. The commands you can use:
+
+```
+oc scale --replicas=0 deploy/machine-api-operator -n openshift-machine-api
+oc edit deploy/machine-api-operator -n openshift-machine-api
+oc scale --replicas=1 deploy/machine-api-operator -n openshift-machine-api
+```
```

Comment:
I think just doing an `oc edit` would suffice no? This would cause the deployment to rolling update the container right?

Reply:
Right, that's a copy-paste from Mike's notes :) I guess it's here for a better understanding of what's going on.

Reply:
this might have also been me being over-cautious. i had thought that the mao was watching it's deployment and preventing me from messing with the values. in hindsight that was probabaly cvo, but this got left in my instructions.